Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,250 @@
+        return is_greater_than(spec_version);
+    }
+    if (compare_op == "<"sv) {
+        return is_lesser_than(spec_version);
+    }
+    if (compare_op == ">="sv) {
+        return is_same(spec_version) || is_greater_than(spec_version);
+    }
+    if (compare_op == "<="sv) {
+        return is_same(spec_version) || is_lesser_than(spec_version);
+    }
+
+    return false;
+}
+
+ErrorOr<SemVer> from_string_view(StringView const& version, char mmp_sep)
+{
+    using namespace regex;
+    if (version.count(mmp_sep) < 2) {
+        return AK::Error::from_string_view("Insufficient version separator"sv);
+    }
+
+    if (version.count('+') > 1) {
+        return AK::Error::from_string_view("Build metadata must be defined once"sv);
+    }
+
+    GenericLexer lexer(version.trim_whitespace());
+    if (lexer.tell_remaining() == 0) {
+        return AK::Error::from_string_view("Version string is empty"sv);
+    }
+
+    auto version_part = lexer.consume_until(mmp_sep).to_uint<u64>();
+    if (!version_part.has_value()) {
+        return AK::Error::from_string_view("Major version is not numeric"sv);
+    }
+    auto v_major = version_part.value();
+
+    lexer.consume();
+
+    version_part = lexer.consume_until(mmp_sep).to_uint<u64>();
+    if (!version_part.has_value()) {
+        return AK::Error::from_string_view("Minor version is not numeric"sv);
+    }
+    auto v_minor = version_part.value();
+
+    lexer.consume();
+
+    version_part = lexer.consume_while<Function<bool(char)>>([](char ch) {
+                            return ch >= '0' && ch <= '9';
+                        })
+                       .to_uint<u64>();
+    if (!version_part.has_value()) {
+        return AK::Error::from_string_view("Patch version is not numeric"sv);
+    }
+    auto v_patch = version_part.value();
+
+    if (lexer.is_eof()) {
+        return SemVer(v_major, v_minor, v_patch, mmp_sep);
+    }
+
+    auto const& suffix_ch = lexer.consume();
+    StringView buildmetadata;
+    StringView prerelease;
+
+    switch (suffix_ch) {
+    case '+':
+        buildmetadata = lexer.consume_all();
+        break;
+    case '-':
+        prerelease = lexer.consume_until('+');
+
+        if (!lexer.is_eof()) {
+            lexer.consume();
+            if (lexer.is_eof()) {
+                return AK::Error::from_string_view("Build metadata is required"sv);
+            }
+            buildmetadata = lexer.consume_all();
+        }
+        break;
+    default:
+        // TODO: Add context information like actual character (peek) and its index
+        // "Expected prerelease (-) or build metadata (+) character at {}. Found {}"
+        return AK::Error::from_string_view("Malformed version syntax. Expected + or - characters"sv);
+    }
+
+    // Validating consumed prerelease and buildmetadata
+    Regex<ECMA262Parser> only_digits_prerelease("^(\\.?\\d+\\.?)(\\d+\\.?)*$");
+    if (prerelease.ends_with('.')
+        || only_digits_prerelease.has_match(prerelease)
```

Comment:
As far as I can tell, the specification doesn't disallow digits-only prerelease specifiers.